src/community/jdbcconfig/src/main/java/org/geoserver/jdbcconfig/internal/ConfigDatabase.java

```diff
@@ -1678,4 +1692,110 @@ public void handleSettingsAdded(SettingsInfo settings) {
             updateCache(settings);
         }
     }
+
+    public class CatalogReferenceUpdater implements CatalogVisitor {
+
+        private CatalogReferenceUpdater() {}
+
+        @Override
+        public void visit(Catalog catalog) {}
+
+        @Override
+        public void visit(WorkspaceInfo workspace) {}
+
+        @Override
+        public void visit(NamespaceInfo workspace) {}
+
+        @Override
+        public void visit(DataStoreInfo dataStore) {
+            if (dataStore.getWorkspace() != null) {
+                dataStore.setWorkspace(
+                        getById(dataStore.getWorkspace().getId(), WorkspaceInfo.class));
+            }
+        }
+
+        @Override
+        public void visit(CoverageStoreInfo coverageStore) {
+            if (coverageStore.getWorkspace() != null) {
+                coverageStore.setWorkspace(
+                        getById(coverageStore.getWorkspace().getId(), WorkspaceInfo.class));
+            }
+        }
+
+        @Override
+        public void visit(WMSStoreInfo wmsStore) {
+            if (wmsStore.getWorkspace() != null) {
+                wmsStore.setWorkspace(
+                        getById(wmsStore.getWorkspace().getId(), WorkspaceInfo.class));
+            }
+        }
+
+        @Override
+        public void visit(FeatureTypeInfo featureType) {
+            if (featureType.getNamespace() != null) {
+                featureType.setNamespace(
+                        getById(featureType.getNamespace().getId(), NamespaceInfo.class));
+            }
+            featureType.setStore(getById(featureType.getStore().getId(), StoreInfo.class));
+        }
+
+        @Override
+        public void visit(CoverageInfo coverage) {
+            if (coverage.getNamespace() != null) {
+                coverage.setNamespace(
+                        getById(coverage.getNamespace().getId(), NamespaceInfo.class));
+            }
+            coverage.setStore(getById(coverage.getStore().getId(), StoreInfo.class));
+        }
+
+        @Override
+        public void visit(LayerInfo layer) {
```

Comment:
Does the Resource also need to be updated?

Reply:
no, there is a separate mechanism for that, the layer gets invalidated if the resource changes